### PR TITLE
Add a new method to set the rocketchat api url

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
This pull request allows to get the rocketchat api service url from the platform settings or use a default value.
## Reviewers 
- [ ] @felipemontoya 
- [ ] @jfavellar90

### CODE REVIEW

File: rocketc/rocketc.py [hunk 1]
```diff
@@ -106,29 +108,39 @@ def init(self):
         else:
             return response['errorType']
 
+    def get_url_api_rocket_chat(self):
+        """
+        This method retunrs the rocketChat url service where someone can acces to API
+        """
+        settings = self.get_xblock_settings()
+        if "url_service" in settings:
+            return "{}/{}/{}".format(settings["url_service"], "api", "v1")
```
Comment (felipemontoya) [inline rocketc/rocketc.py]: You could make it like `"/".join(...)` if you want

File: rocketc/rocketc.py [hunk 2]
```diff
@@ -106,29 +108,39 @@ def init(self):
...
+
+        return "{}/{}/{}".format("http://localhost:3000", "api", "v1")
```
Comment (felipemontoya) [inline rocketc/rocketc.py]: should this also be a setting? perhaps in the local settings

File: rocketc/rocketc.py [hunk 3]
```diff
@@ -106,29 +108,39 @@ def init(self):
...
         user_data["anonymous_student_id"] = runtime.anonymous_student_id
-        self.user_data = user_data # pylint: disable=attribute-defined-outside-init
+        self.user_data = user_data  # pylint: disable=attribute-defined-outside-init
 
     def get_admin_data(self):
         """
         This method initializes admin's authToken and userId
         """
-        url = "{}/{}".format(URL_PREFIX, "login")
+        url = "{}/{}".format(self.url_api_rocket_chat, "login")
```
Comment (felipemontoya) [inline rocketc/rocketc.py]: the API Url is something like: `http://localhost:3000/api/v1`. This means, the result to url here is `http://localhost:3000/api/v1/login`. Correct?

File: rocketc/tests/test_rocketChat.py [hunk 4]
```diff
@@ -13,6 +13,7 @@ def setUp(self):
         self.block = RocketChatXBlock(
             self.runtime_mock, scope_ids=scope_ids_mock)
         self.block.admin_data = MagicMock()
+        self.block.url_api_rocket_chat = MagicMock ()
```
Comment (felipemontoya) [inline rocketc/tests/test_rocketChat.py]: extra space

File: rocketc/rocketc.py (hunk 3 above)
Comment (andrey-canon) [reply rocketc/rocketc.py]: yes, the result is that

File: rocketc/rocketc.py [hunk 5]
```diff
@@ -109,29 +111,39 @@ def init(self):
...
 
+    def get_url_api_rocket_chat(self):
+        """
+        This method retunrs the rocketChat url service where someone can acces to API
+        """
+        xblock_settings = self.get_xblock_settings()
+        if "url_service" in xblock_settings:
+            return "/".join([xblock_settings["url_service"], "api", "v1"])
+
+        return "/".join(["http://localhost:3000", "api", "v1"])
```
Comment (jfavellar90) [inline rocketc/rocketc.py]: will this eventually changed by a setting? the base rocket chat url?

Comment (jfavellar90) [reply rocketc/rocketc.py]: Sorry ignore this comment, this is just a fallback if the setting is not defined

**Pull request merged**